src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/AccessToken/OidcUserInfoTokenHandlerFactory.php

```diff
@@ -27,23 +26,8 @@ class OidcUserInfoTokenHandlerFactory implements TokenHandlerFactoryInterface
     public function create(ContainerBuilder $container, string $id, array|string $config): void
     {
         $tokenHandlerDefinition = $container->setDefinition($id, new ChildDefinition('security.access_token_handler.oidc_user_info'));
+        $tokenHandlerDefinition->replaceArgument(0, new Reference($config['client']));
         $tokenHandlerDefinition->replaceArgument(2, $config['claim']);
-
-        // Create the client service
-        if (!isset($config['client']['id'])) {
-            $clientDefinitionId = 'http_client.security.access_token_handler.oidc_user_info';
-            if (!ContainerBuilder::willBeAvailable('symfony/http-client', HttpClient::class, ['symfony/security-bundle'])) {
-                $container->register($clientDefinitionId, 'stdClass')
-                    ->addError('You cannot use the "oidc_user_info" token handler since the HttpClient component is not installed. Try running "composer require symfony/http-client".');
```

Comment:
we might still need to register this special def when the client is not installed, isn't it?

Reply:
`client` option receives an http_client service id, so I guess the user installed symfony/http-client to declare it. Do we need to double-check it here?

Reply:
Solved with the addition of `base_uri` option